scripts/update_deps.py

```diff
@@ -44,15 +44,32 @@
 
 
 class PEP503Parser(HTMLParser):
-  """Parser for PEP 503 simple repository API pages."""
+  """Parser for PEP 503 simple repository API pages.
+
+  This parser extracts all links and their link text from the
+  HTML content of a PEP 503 index page.
+  """
+
+  links: list[tuple[str, str]]
+  """List of (href, text) tuples for all links found."""
+
+  current_link: str | None
```

Comment:
Nit: make this private as it's not public the class's public API (unlike `links`)?